### PR TITLE
Reduce frequency of AdsLinks checks.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #9141

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by benbowler — APPROVED** [review]
Looking good, and tested, thanks @ankitrox. Moving to MR.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @ankitrox ! This one is almost there, but we can simplify this a fair bit with an alternate approach.

File: includes/Modules/Analytics_4/Synchronize_AdsLinked.php [hunk 1]
```diff
@@ -117,13 +135,17 @@ protected function synchronize_ads_linked_status() {
 	 * @since 1.124.0
 	 */
 	public function maybe_schedule_synchronize_ads_linked() {
+		if ( is_admin() && 'googlesitekit-dashboard' === $this->context->input()->filter( INPUT_GET, 'page' ) ) {
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (ankitrox) [reply includes/Modules/Analytics_4/Synchronize_AdsLinked.php]: Thank you @aaemnnosttv . I have made this change.

File: includes/Modules/Analytics_4/Synchronize_AdsLinked.php [hunk 2]
```diff
@@ -40,17 +42,27 @@ class Synchronize_AdsLinked {
...
+	protected $context;
+
 	/**
 	 * Constructor.
 	 *
 	 * @since 1.124.0
 	 *
 	 * @param Analytics_4  $analytics_4  Analytics 4 instance.
 	 * @param User_Options $user_options User_Options instance.
+	 * @param Context      $context      Context instance.
```
[the remark this answers is not in the document]
Comment (ankitrox) [reply includes/Modules/Analytics_4/Synchronize_AdsLinked.php:60]: Noted.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
@ankitrox Sorry if this wasn't clear, but given that the added `Context` instance is no longer necessary since we aren't accessing query parameters, those related additions/changes here should be reverted. Thanks!

**Comment by ankitrox** [comment]
> @ankitrox Sorry if this wasn't clear, but given that the added `Context` instance is no longer necessary since we aren't accessing query parameters, those related additions/changes here should be reverted. Thanks!

Hi @aaemnnosttv ,

Sorry for the confusion, I did not get it at the first place.

You are right, `$context` is no longer required, so I have removed it and pushed the changes.

Thanks

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @ankitrox !

**Pull request merged**